### ISSUE TITLE
Opencv dnn is inconsistent with tensorflow detection results.

### ISSUE THREAD

**Issue by x12901** [opencv/opencv#18464]
##### System information (version)
- OpenCV => 4.4
- Operating System / Platform => Windows10 64 Bit
- Opencv-contrib-python=>4.4.0.44
- Opencv-python =>4.4.0.44
- Tensorflow=>1.15
- Tensorflow_models=>r1.12.0
##### Detailed description
I trained my model with TensorFlow Object Detection API.The model is based on the pre-trained model mask_rcnn_inception_v2_coco_2018_01_28.
https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf1_detection_zoo.md
Use tf_text_graph_mask_rcnn.py which generate a text graph representation for a frozen .pb model depends on it's architecture.Then I ran the example code.
https://github.com/opencv/opencv/blob/master/samples/dnn/mask_rcnn.py
The detection results are inconsistent with using tensorflow api.I've tried other code, but that doesn't seem to work either.
https://github.com/spmallick/learnopencv/blob/master/Mask-RCNN/mask_rcnn.cpp
I'm guessing it has something to do with my label.pbtxt, my project only has 3 classes and the pre-trained model config has 90 classes. Or maybe it has something to do with something else, I was hoping you could give me some advice.
